presidio-analyzer/tests/test_spacy_recognizer.py

```diff
@@ -9,85 +10,181 @@
+    
+    # Bug #617 : Spacy Recognizer doesn't recognize Mr. Tailor as PERSON even though online spacy visualizer indicates that it does
+    # See http://textanalysisonline.com/spacy-named-entity-recognition-ner
+    # def test_person_title_with_last_name_with_context_and_time(self):
+    #     name = 'Mr. Tailor'
+    #     context = 'Good morning'
+    #     results = spacy_recognizer.analyze('{} {}'.format(context, name), entities)
+
+    #     assert len(results) == 2
+    #     assert_result_within_score_range(results[1], entities[1], 5, 12, NER_STRENGTH, 1)
+    #     assert_result_within_score_range(results[0], entities[0], 17, 23, NER_STRENGTH, 1)
 
     def test_person_full_middle_name(self):
         name = 'Richard Milhous Nixon'
         results = spacy_recognizer.analyze(name, entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 0
-        assert results[0].end == 21
+        assert_result(results[0], entities[0], 0, 21, NER_STRENGTH)
 
-    def test_person_full_middle_letter_name(self):
+    def test_person_full_name_with_middle_letter(self):
         name = 'Richard M. Nixon'
         results = spacy_recognizer.analyze(name, entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 0
-        assert results[0].end == 16
+        assert_result(results[0], entities[0], 0, 16, NER_STRENGTH)
 
     def test_person_full_name_complex(self):
         name = 'Richard (Ric) C. Henderson'
         results = spacy_recognizer.analyze(name, entities)
 
         assert len(results) == 1
-        assert results[0].score >= NER_STRENGTH
-        assert results[0].entity_type == entities[0]
-        assert results[0].start == 0
-        assert results[0].end == 26
+        assert_result(results[0], entities[0], 0, 26, NER_STRENGTH)
+
+    def test_person_last_name_is_also_a_date_expected_person_only(self):
+        name = 'Dan May'
+        results = spacy_recognizer.analyze(name, entities)
+
+        assert len(results) == 1
+        assert_result(results[0], entities[0], 0, 7, NER_STRENGTH, )
+    
+    # Bug #617 : Spacy Recognizer doesn't recognize Mr. Tailor as PERSON even though online spacy demo indicates that it does
+    # See http://textanalysisonline.com/spacy-named-entity-recognition-ner
+    def test_person_last_name_is_also_a_date_with_context_expected_person_only(self):
+        name = 'Dan May'
+        context = "has a bank account"
+        results = spacy_recognizer.analyze('{} {}'.format(name, context), entities)
+
+        assert len(results) == 1
+        assert_result_within_score_range(results[0], entities[0], 0, 7, NER_STRENGTH, 1)
+
+    # Bug #617 : Spacy Recognizer doesn't recognize Mr. Tailor as PERSON even though online spacy demo indicates that it does
```

Comment:
These tests expect to pass but fail? I suggest adding them only when replacing spaCy. otherwise it will forever remain commented out

Reply:
Those texts are failing, hence I opened bug 617.
It's ok to have them here commented out, as long as there's an open bug associated to them. 
It's a common practice. They were actually created in a TDD approach, and helped identifying the bug.